### DISCUSSION TITLE
Documentation on methods that return closures so you can grok what sort of thing the closure is expected to return or do.

### DISCUSSION THREAD

**Discussion by patrickcurl** [filamentphp/filament#4192] [Ideas]
Example: 

I'm creating a Location service that calls nominatim.openstreetmap.org/search. 

I'm trying to match this up w/ a location record using a hash from the coords and storing them in the cache.

the getOptionLabelFromRecordUsing method in my Resource class is very limited and I have to dig pretty deep to figure out what's going on to get it working. 

I think param/return docs are definitely overkill but ide development will be a lot better w/ some better docs. 

Ideally, it'd be nice for good docs on every method being opensource, mainly for ide completions where it shows you the method call and phpdoc, but the ones w/ closures especially.

**Comment by danharrin** [comment]
Apart from the method-specific parameters which are all mentioned in the docs, all of the form component methods accept the same set of parameters, absolutely everywhere: https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation

But yes, for the other packages you are absolutely correct, these parameters still need documenting.
